test/ReactViews/Workbench/Controls/LegendSpec.tsx

```diff
@@ -0,0 +1,51 @@
+const findAllWithType = require("react-shallow-testutils").findAllWithType;
+import { getShallowRenderedOutput } from "../../MoreShallowTools";
+import React from "react";
+
+import Terria from "../../../../lib/Models/Terria";
+import WebMapServiceCatalogItem from "../../../../lib/Models/WebMapServiceCatalogItem";
+import Legend from "../../../../lib/ReactViews/Workbench/Controls/Legend";
+
+describe("Legend", function() {
+  let terria: Terria;
+  let wmsItem: WebMapServiceCatalogItem;
+
+  beforeEach(function() {
+    terria = new Terria({
+      baseUrl: "./"
+    });
+
+    wmsItem = new WebMapServiceCatalogItem("mywms", terria);
+    wmsItem.setTrait(
```

Comment:
Wrap this in action?

Reply:
How come? setTrait is already an action itself, right?

Reply:
Ah you're right! 
That was a more recent change I didn't realize. Thanks!